File: Java/1865 - Mjölnir.java
import java.util.Scanner;
import java.io.IOException;

public class Main {
 
    public static void main(String[] args) throws IOException {
 
        int casos;
        String valor;
        
        Scanner entrada = new Scanner(System.in);
        
        casos = entrada.nextInt();
        entrada.nextLine();
        
       for(int i = 0; i < casos; i++){
            
            valor = entrada.nextLine();
            
            if(valor.toLowerCase().contains("Thor".toLowerCase())){
                System.out.println("Y");
            }else{
                System.out.println("N");
            }
            
        }
 
    }
 
}

File: C++/1865 - Mjölnir.cpp
#include <stdio.h>
#include <iostream>
#include <string>

using namespace std;

int main() {
    
    int casos = 0;
    string valor = "";
    string thor = "Thor";
    
    cin>>casos;
    
    for(int i = 0; i < casos; i++){
        cin>>valor;
        
        if(valor.compare(thor) != 0){
            cout<<"N"<<endl;
        }else{
            cout<<"Y"<<endl;
        }
        
        cin>>valor;
    }
    
    return 0;
    
}
